### PR TITLE
[bitnami/external-dns] PDB fixes

### LINKED ISSUE #26199: [external-dns] Default PDB is incompatible with default replica count
### Name and Version

bitnami/external-dns 7.4.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Install chart with defaults

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

Maintenance to infrastructure is possible

### What do you see instead?

Maintenance to infrastructure is blocked because default configuration of external-dns blocks pod migration due to minAvailable 1 and replica 1.

### Additional information

Introduced in https://github.com/bitnami/charts/pull/25933

### PR DESCRIPTION
**Pull request by fmulero**
### Description of the change

Remove default value for `pdb.minAvailable` and set `maxUnavailable`to 1 in PodDisruptionBudgets if none is set.

### Benefits

Maintenance to infrastructure is allowed with chart defaults.

### Possible drawbacks

Some users could expect old value in `minAvailable`. In that case the value should be set explicitly.

### Applicable issues

- fixes #26199

### Additional information

Follow up #25933

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by juan131 — CHANGES_REQUESTED** [review]
We should keep 1 replica by default and set `pdb.maxUnavailable` instead.

```yaml
pdb:
  minAvailable: ""
  maxUnavailable: 1
```

**Comment by fmulero** [comment]
> We should keep 1 replica by default and set `pdb.maxUnavailable` instead.
> 
> ```yaml
> pdb:
>   minAvailable: ""
>   maxUnavailable: 1
> ```

With those defaults, if any user tries to install the chart setting only `pdb.minAvailable=1' this error will be faced:
```console
$ helm install test bitnami/external-dns --set pdb.minAvailable=1
Error: INSTALLATION FAILED: PodDisruptionBudget.policy "test-external-dns" is invalid: spec: Invalid value: policy.PodDisruptionBudgetSpec{MinAvailable:(*intstr.IntOrString)(0xc00d46e580), Selector:(*v1.LabelSelector)(0xc00d46e5a0), MaxUnavailable:(*intstr.IntOrString)(0xc00d46e560), UnhealthyPodEvictionPolicy:(*policy.UnhealthyPodEvictionPolicyType)(nil)}: minAvailable and maxUnavailable cannot be both set
```
Users should unset `pdb.maxUnavailable` explicitly. 

With the current approach that is not needed. Users settings take precedence but  if `pdb.maxUnavailable` and `pdb.minAvailable` are not set , `maxUnavailable` will be set to 1 by default in the PodDisruptionBudget.

File: bitnami/external-dns/templates/pdb.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ spec:
   {{- if .Values.pdb.minAvailable }}
   minAvailable: {{ .Values.pdb.minAvailable }}
   {{- end }}
-  {{- if .Values.pdb.maxUnavailable }}
-  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- if or .Values.pdb.maxUnavailable (not .Values.pdb.minAvailable) }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable | default 1 }}
   {{- end }}
```
Comment (juan131) [inline bitnami/external-dns/templates/pdb.yaml:25]: Nice! I love this workaround, it'll definitely help with the UX. That said, please ensure we state in the `pdb.maxUnavailable` parameter description this new behavior

**Review by juan131 — DISMISSED** [review]
LGTM

**Pull request merged**